src/integrations/admin/indexing-notification-integration.php

```diff
@@ -0,0 +1,245 @@
+
+	/**
+	 * Represents the reason that the indexing process failed and should be tried again.
+	 */
+	const REASON_INDEXING_FAILED = 'indexing_failed';
+
+	/**
+	 * Represents the reason that the permalink settings are changed.
+	 */
+	const REASON_PERMALINK_SETTINGS = 'permalink_settings_changed';
+
+	/**
+	 * Represents the reason that the category base is changed.
+	 */
+	const REASON_CATEGORY_BASE_PREFIX = 'category_base_changed';
+
+	/**
+	 * Represents the reason that the home url option is changed.
+	 */
+	const REASON_HOME_URL_OPTION = 'home_url_option_changed';
+
+	/**
+	 * The indexing integration.
+	 *
+	 * @var Indexing_Integration
+	 */
+	private $indexing_integration;
+
+	/**
+	 * The Yoast notification center.
+	 *
+	 * @var Yoast_Notification_Center
+	 */
+	private $notification_center;
+
+	/**
+	 * The options helper.
+	 *
+	 * @var Options_Helper
+	 */
+	private $options_helper;
+
+	/**
+	 * The product helper.
+	 *
+	 * @var Product_Helper
+	 */
+	private $product_helper;
+
+	/**
+	 * Prominent_Words_Notifier constructor.
+	 *
+	 * @param Indexing_Integration      $indexing_integration The indexing integration.
+	 * @param Yoast_Notification_Center $notification_center  The notification center.
+	 * @param Options_Helper            $options_helper       The options helper.
+	 * @param Product_Helper            $product_helper       The product helper.
+	 */
+	public function __construct(
+		Indexing_Integration $indexing_integration,
+		Yoast_Notification_Center $notification_center,
+		Options_Helper $options_helper,
+		Product_Helper $product_helper
+	) {
+		$this->indexing_integration = $indexing_integration;
+		$this->notification_center  = $notification_center;
+		$this->options_helper       = $options_helper;
+		$this->product_helper       = $product_helper;
+	}
+
+	/**
+	 * Initializes the integration.
+	 *
+	 * Adds hooks and jobs to cleanup or add the notification when necessary.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		if ( \filter_input( INPUT_GET, 'page' ) === 'wpseo_dashboard' ) {
+			\add_action( 'admin_init', [ $this, 'cleanup_notification' ] );
+		}
+
+		if ( ! \wp_next_scheduled( self::NOTIFICATION_ID ) ) {
+			\wp_schedule_event( time(), 'daily', self::NOTIFICATION_ID );
+		}
+
+		\add_action( self::NOTIFICATION_ID, [ $this, 'create_notification' ] );
+	}
+
+	/**
+	 * Returns the conditionals based in which this loadable should be active.
+	 *
+	 * @return array
+	 */
+	public static function get_conditionals() {
+		return [
+			Admin_Conditional::class,
+		];
+	}
+
+	/**
+	 * Checks whether the notification should be shown and removes
+	 * it from the notification center if this is the case.
+	 */
+	public function create_notification() {
+		$this->notification_center->add_notification( $this->notification() );
+		$notification = $this->notification_center->get_notification_by_id( self::NOTIFICATION_ID );
+
+		if ( $notification || ! $this->should_show_notification() ) {
```

Comment:
Shouldn't this be `! $notificiation`? Currently, if there is a notification, you return.

Reply:
We want to make sure that a notification is not added multiple times.
In the line above, we check whether the notification center already has the notification, if so, we return right here.

Reply:
For extra context: the notifications are stored in the database between page requests.